### PR TITLE
apiv2 add bindings for logs|events

### PR DESCRIPTION
**Pull request by baude**
add go-bindings for logs and events. tests were also added.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/bindings/containers/logs.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	"encoding/binary"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/containers/libpod/pkg/bindings"
+)
+
+// Logs obtains a container's logs given the options provided.  The logs are then send to the
```
Comment (TomSweeneyRedHat) [inline pkg/bindings/containers/logs.go]: nit, ignore unless you get other comments.  send -> sent

File: pkg/bindings/containers/logs.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+// stdout|stderr channels as strings.
+func Logs(ctx context.Context, nameOrID string, follow *bool, opts LogOptions, stdoutChan, stderrChan chan string) error {
+	conn, err := bindings.GetClient(ctx)
+	if err != nil {
+		return err
+	}
+	params := url.Values{}
+	if follow != nil {
```
Comment (TomSweeneyRedHat) [inline pkg/bindings/containers/logs.go]: just curious why follow isn't in the opts too?

File: pkg/bindings/test/containers_test.go [hunk 3]
```diff
@@ -357,4 +357,27 @@ var _ = Describe("Podman containers ", func() {
...
+		opts := containers.LogOptions{Stdout: &bindings.PTrue}
+		go func() {
+			err := containers.Logs(bt.conn, r.ID, &bindings.PTrue, opts, stdoutChan, nil)
+			Expect(err).To(BeNil())
+		}()
+		o := <-stdoutChan
+		o = strings.ReplaceAll(o, "\r", "")
+		_, err = time.Parse(time.RFC1123Z, o)
+		Expect(err).To(BeNil())
+	})
```
Comment (TomSweeneyRedHat) [inline pkg/bindings/test/containers_test.go]: It would be good to have a follows set to False test too.

File: pkg/bindings/containers/logs.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+	"github.com/containers/libpod/pkg/bindings"
+)
+
+// Logs obtains a container's logs given the options provided.  The logs are then sent to the
+// stdout|stderr channels as strings.
+func Logs(ctx context.Context, nameOrID string, opts LogOptions, stdoutChan, stderrChan chan string) error {
+	conn, err := bindings.GetClient(ctx)
+	if err != nil {
+		return err
```
Comment (rhatdan) [inline pkg/bindings/containers/logs.go]: Should this error be wrapped?

File: pkg/bindings/containers/logs.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	if opts.Until != nil {
+		params.Set("until", *opts.Until)
+	}
+	// The API requires either stdout|stderr be used. If neither are specified, we specify stdout
+	if opts.Stdout == nil && opts.Stderr == nil {
+		params.Set("stdout", strconv.FormatBool(true))
+	}
+	response, err := conn.DoRequest(nil, http.MethodGet, "/containers/%s/logs", params, nameOrID)
+	if err != nil {
+		return err
```
Comment (rhatdan) [inline pkg/bindings/containers/logs.go]: Should this error be wrapped?

File: pkg/bindings/containers/logs.go [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+	// first byte determines stderr=2|stdout=1
+	// bytes 4-7 len(msg) in uint32
+	for {
+		stream, msgSize, err := readHeader(response.Body)
+		if err != nil {
+			// In case the server side closes up shop because !follow
+			if err == io.EOF {
+				break
+			}
+			return err
```
Comment (rhatdan) [inline pkg/bindings/containers/logs.go]: Should this error be wrapped?

File: pkg/bindings/containers/logs.go [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+		}
+		msg, err := readMsg(response.Body, msgSize)
+		if err != nil {
+			return err
```
Comment (rhatdan) [inline pkg/bindings/containers/logs.go]: Should this error be wrapped?

File: pkg/bindings/system/system.go [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	"github.com/sirupsen/logrus"
+)
+
+// Events allows you to monitor libdpod related events like container creation and
+// removal.  The events are then passed to the eventChan provided. The optional cancelChan
+// can be used to cancel the read of events and close down the HTTP connection.
+func Events(ctx context.Context, eventChan chan (handlers.Event), cancelChan chan bool, since, until *string, filters map[string][]string) error {
+	conn, err := bindings.GetClient(ctx)
+	if err != nil {
+		return err
```
Comment (rhatdan) [inline pkg/bindings/system/system.go]: Should this error be wrapped?

File: pkg/bindings/system/system.go [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+	if since != nil {
+		params.Set("since", *since)
+	}
+	if until != nil {
+		params.Set("until", *until)
+	}
+	if filters != nil {
+		filterString, err := bindings.FiltersToString(filters)
+		if err != nil {
+			return err
```
Comment (rhatdan) [inline pkg/bindings/system/system.go]: Should this error be wrapped?

File: pkg/bindings/system/system.go [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+		}
+		params.Set("filters", filterString)
+	}
+	response, err := conn.DoRequest(nil, http.MethodGet, "/events", params)
+	if err != nil {
+		return err
```
Comment (rhatdan) [inline pkg/bindings/system/system.go]: Should this error be wrapped?

File: pkg/bindings/system/system.go [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+	}
+	if cancelChan != nil {
+		go func() {
+			<-cancelChan
+			err = response.Body.Close()
+			logrus.Error(err)
```
Comment (rhatdan) [inline pkg/bindings/system/system.go]: Should this error be wrapped?

File: pkg/bindings/system/system.go [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+		}()
+	}
+	dec := json.NewDecoder(response.Body)
+	for {
+		e := handlers.Event{}
+		if err := dec.Decode(&e); err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
```
Comment (rhatdan) [inline pkg/bindings/system/system.go]: Should this error be wrapped?

**Pull request merged**